docs/ci-setup.md

```diff
@@ -0,0 +1,36 @@
+## CI Setup
+
+Setting up appium in an automated environment brings a few challenges with it.
+Any scenario where user interaction is required must be automated or avoided all together.
+For real device set up you should first follow the steps in [this guide](https://github.com/appium/appium-xcuitest-driver/blob/master/docs/real-device-config.md).
+
+### Keychains
+
+One common such scenario is a prompt asking for a keychain to be unlocked in order to sign the WebdriverAgent.
+There are multiple possible solutions for this:
+
+1. Keychains can be set to have no timeout and be unlocked manually once. This can be done using the keychain access application. Sometimes keychains still unlock themselves though and this approach is not recommended.
+2. [It is possible to create a second keychain](https://github.com/appium/appium-xcuitest-driver/blob/master/README.md#real-device-security-settings), which just stores the required certificate to sign the WebdriverAgent.
+The issue with this approach is, that Codesign wants to unlock all listed keychains regardless of the specified keychain, thus leading to a password prompt.
+This can be avoided by setting the default keychain and basically hiding the login keychain at the start of building.
+https://stackoverflow.com/questions/16550594/jenkins-xcode-build-works-codesign-fails
+Not practically when running other build jobs simultaneously.
+3. Stick with the existing keychains as in approach number one, but explicitly call unlock keychain before **each** build. This can be done using fastlane `unlock_keychain` or by using `security unlock-keychain` directly.
```

Comment:
it would probably be useful to add links to these both CLI utilities

Reply:
I couldn't find any official resources for the security CLI. It can be executed with a help parameter though to get at least some documentation.

Reply:
maybe https://www.unix.com/man-page/osx/1/security/ ?

Reply:
Good find! I replaced the link